app/scripts/modules/core/src/pipeline/config/preconditions/selector/preconditionSelector.directive.js

```diff
@@ -45,13 +46,24 @@ module.exports = angular.module('spinnaker.core.pipeline.config.preconditions.se
       return preconditionConfig ? preconditionConfig.contextTemplateUrl : '';
     };
 
+    this.clusterChanged = function (clusterName) {
+        let clusterFilter = appListExtractorService.monikerClusterNameFilter(clusterName);
+        let moniker = _.first(_.uniq(appListExtractorService.getMonikers([$scope.application], clusterFilter)));
```

Comment:
Should import `first` and `uniq` from lodash, too, and not rely on `_` being globally available...

Since you're only using the first entry in the array, it probably makes sense to skip the `uniq` call.